easybuild/main.py

```diff
@@ -590,10 +468,198 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
+        # for path in orig_paths:
+        #     validate_command_opts(args, opts_per_ec[path])
+
+        # Loop over each item in the EasyStack file, each time updating the config
+        # This is because each item in an EasyStack file can have options associated with it
+        do_cleanup = True
+        for path in orig_paths:
+            _log.debug("Starting build for %s" % path)
+            # NOTE: not sure if this is needed. Is the EasyConfigs cache preserved throughout loops of this iteration?
+            # Current 'path' may have different options associated with it. Thus, resolution of EasyConfigs
+            # to full paths should not be read from cache, but redetermined. Thus, we wipe the cache
+            # wipe_cache()  # Wipes the easyconfig cache with _easyconfigs_cache.clear()  # TODO: implement
+
+            # If EasyConfig specific arguments were supplied in EasyStack file
+            # merge arguments with original command line args
+            if path in opts_per_ec:
+                _log.debug("EasyConfig specific options have been specified for "
+                           "%s in the EasyStack file: %s" % (path, opts_per_ec[path]))
+                if args is None:
+                    args = sys.argv[1:]
+                ec_args = dict_to_argslist(opts_per_ec[path])
+                # By appending ec_args to args, ec_args take priority
+                new_args = args + ec_args
+                _log.info("Argument list for %s after merging command line arguments with EasyConfig specific "
+                          "options from the EasyStack file: %s" % (path, new_args))
+            else:
+                # If no EasyConfig specific arguments are defined, sse original args.
+                # That way,set_up_configuration restores the original config
+                new_args = args
+            eb_go, cfg_settings = set_up_configuration(args=new_args, logfile=logfile, testing=testing,
+                                                       reconfigure=True)
+
+            hooks = load_hooks(options.hooks)
+            do_cleanup &= process_eb_args([path], eb_go.options, cfg_settings, modtool, testing, init_session_state,
+                                          hooks, do_build)
+
+        # Loop done. If overall_success is not false, cleanup
+        # if overall_success or overall_success is None:
+        #     cleanup(logfile, eb_tmpdir, testing)
+    else:
+        do_cleanup = process_eb_args(orig_paths, options, cfg_settings, modtool, testing, init_session_state, hooks,
+                                     do_build)
     # stop logging and cleanup tmp log file, unless one build failed (individual logs are located in eb_tmpdir)
     stop_logging(logfile, logtostdout=options.logtostdout)
-    if overall_success:
-        cleanup(logfile, eb_tmpdir, testing)
+    if do_cleanup:
+        print("Running cleanup")
```

Comment:
please remove this `print` again, probably leftover of debugging

Reply:
Done

Reply:
d98aa4673f9b7c2d46d49e39d6d336e0b5c8e7fe